timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureLockWatchingResource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import java.util.function.Function;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.timelock.api.LockWatchRequest;
+import com.palantir.atlasdb.timelock.lock.watch.ConjureLockWatchingService;
+import com.palantir.atlasdb.timelock.lock.watch.ConjureLockWatchingServiceEndpoints;
+import com.palantir.atlasdb.timelock.lock.watch.UndertowConjureLockWatchingService;
+import com.palantir.conjure.java.undertow.lib.UndertowService;
+import com.palantir.tokens.auth.AuthHeader;
+
+public final class ConjureLockWatchingResource implements UndertowConjureLockWatchingService {
+    private final ConjureResourceExceptionHandler exceptionHandler;
+    private final Function<String, AsyncTimelockService> timelockServices;
+
+    private ConjureLockWatchingResource(
+            RedirectRetryTargeter redirectRetryTargeter,
+            Function<String, AsyncTimelockService> timelockServices) {
+        this.exceptionHandler = new ConjureResourceExceptionHandler(redirectRetryTargeter);
+        this.timelockServices = timelockServices;
+    }
+
+    public static UndertowService undertow(
```

Comment:
sad that we're gonna have to do this for each resource :(

Reply:
It's a nuisance, yeah